src/arg_parsers/filename.rs

```diff
@@ -306,49 +351,41 @@ fn split_off_bracketed_suffix(input: &OsStr) -> Option<(OsString, OsString)> {
 }
 
 /// Parses ImageMagick `format:path`-style argument.
-pub fn parse_path_and_format(input: &OsStr) -> (OsString, Option<ImageFormat>) {
-    // Helper that returns the result in a slightly different shape
-    fn inner(input: &OsStr) -> Option<(OsString, ImageFormat)> {
-        #[cfg(any(unix, target_os = "wasi"))]
-        {
-            let bytes = input.as_bytes(); // From std::os::unix::ffi::OsStrExt
-            let mut iter = bytes.splitn(2, |&b| b == b':');
-            let prefix = iter.next().unwrap();
-            let suffix = iter.next()?;
-            Some((
-                OsStr::from_bytes(suffix).to_owned(), // From std::os::unix::ffi::OsStrExt
-                ImageFormat::from_extension(str::from_utf8(prefix).ok()?)?,
-            ))
-        }
-        #[cfg(windows)]
-        {
-            let wide_chars: Vec<u16> = input.encode_wide().collect(); // From std::os::windows::ffi::OsStrExt
-            let mut iter = wide_chars.splitn(2, |&wc| wc == b':' as u16);
-            let prefix = iter.next().unwrap();
-            // On Windows, ImageMagick treats "c:..." as a path
-            if prefix.len() == 1
-                && u8::try_from(prefix[0]).map(|c| c.is_ascii_alphabetic()) == Ok(true)
-            {
-                return None;
-            }
-            let suffix = iter.next()?;
-            Some((
-                OsString::from_wide(suffix), // From std::os::windows::ffi::OsStringExt
-                ImageFormat::from_extension(String::from_utf16(prefix).ok()?)?,
-            ))
-        }
-        #[cfg(not(any(unix, windows, target_os = "wasi")))]
+pub fn parse_path_and_format(input: &OsStr) -> Option<(OsString, ImageFormat)> {
```

Comment:
This is just moving everything in the inner function to the outer one, because it turns out the return type is already more convenient for the callers.